src/disk_job_pool.cpp

```diff
@@ -44,11 +44,15 @@ namespace aux {
 		, m_job_pool(sizeof(disk_io_job))
 	{}
 
+#if 0
 	disk_job_pool::~disk_job_pool()
 	{
 // #error this should be fixed!
 //		TORRENT_ASSERT(m_jobs_in_use == 0);
 	}
+#else
+	disk_job_pool::~disk_job_pool() = default;
+#endif
```

Comment:
I don't think this is a reasonable change

Reply:
Interestingly, after this change clang-tidy complains that this is trivially destructible. The fix there being to move = default to the header.